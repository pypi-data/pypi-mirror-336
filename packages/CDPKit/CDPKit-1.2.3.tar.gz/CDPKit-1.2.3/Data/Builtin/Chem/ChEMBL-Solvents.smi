WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
