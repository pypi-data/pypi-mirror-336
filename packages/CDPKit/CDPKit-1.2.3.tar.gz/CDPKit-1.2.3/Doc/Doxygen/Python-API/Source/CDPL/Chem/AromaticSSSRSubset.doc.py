#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the extraction of the aromatic rings in the SSSR of a molecular graph.
# 
class AromaticSSSRSubset(FragmentList):

    ##
    # \brief Constructs an empty <tt>AromaticSSSRSubset</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Initializes a copy of the \e %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \e %AromaticSSSRSubset instance to copy.
    # 
    def __init__(arom_sssr: AromaticSSSRSubset) -> None: pass

    ##
    # \brief Construct a <tt>AromaticSSSRSubset</tt> instance that contains all aromatic rings of the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \c %AromaticSSSRSubset instance to copy.
    # \return \a self
    # 
    def assign(arom_sssr: AromaticSSSRSubset) -> AromaticSSSRSubset: pass

    ##
    # \brief Replaces the current set of rings by the aromatic rings in the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def extract(molgraph: MolecularGraph) -> None: pass
