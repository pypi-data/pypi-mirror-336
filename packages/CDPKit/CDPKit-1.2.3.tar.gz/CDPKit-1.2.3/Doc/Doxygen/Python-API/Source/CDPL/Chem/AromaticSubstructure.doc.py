#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the perception of aromatic atoms and bonds in a molecular graph.
# 
class AromaticSubstructure(Fragment):

    ##
    # \brief Constructs an empty <tt>AromaticSubstructure</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Construct a <tt>AromaticSubstructure</tt> instance that consists of the aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the currently stored atoms and bonds by the set of aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def perceive(molgraph: MolecularGraph) -> None: pass
