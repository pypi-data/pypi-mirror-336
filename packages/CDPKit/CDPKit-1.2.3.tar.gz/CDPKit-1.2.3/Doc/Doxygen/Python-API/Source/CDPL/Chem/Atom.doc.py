#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Atom.
# 
class Atom(AtomContainer, BondContainer, Entity3D):

    ##
    # \brief 
    #
    class AtomSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Atom: pass

        ##
        # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
        # \param atom The value to test for membership.
        # \return The result of the membership test operation.
        # 
        def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief 
    #
    class BondSequence(Boost.Python.instance):

        ##
        # \brief 
        # \return 
        #
        def __len__() -> int: pass

        ##
        # \brief 
        # \param idx 
        # \return 
        #
        def __getitem__(idx: int) -> Bond: pass

        ##
        # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
        # \param bond The value to test for membership.
        # \return The result of the membership test operation.
        # 
        def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Initializes the \e %Atom instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Returns a reference to the parent molecule.
    # 
    # \return A reference to the parent molecule.
    # 
    def getMolecule() -> Molecule: pass

    ##
    # \brief Returns a reference to the Chem.Bond object that connects this atom to the argument atom.
    # 
    # \param atom The adjacent atom for which to return the connecting bond.
    # 
    # \return A reference to the Chem.Bond object connecting this atom and the argument atom. 
    # 
    # \throw Base.ItemNotFound if the argument atom is not connected to this atom.
    # 
    def getBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief Returns a reference to the Chem.Bond object that connects this atom to the argument atom.
    # 
    # \param atom The adjacent atom for which to return a reference to the connecting bond.
    # 
    # \return A reference to the Chem.Bond object that connects this atom to the argument atom, or <em>None</em> if the argument atom is not connected.
    # 
    def findBondToAtom(atom: Atom) -> Bond: pass

    ##
    # \brief Returns the index of the atom in its parent molecule.
    # 
    # \return The zero-based index of the atom.
    # 
    def getIndex() -> int: pass

    ##
    # \brief Assignment operator that replaces the current set of properties with the properties of <em>atom</em>;.
    # 
    # \param atom The atom whose properties get copied.
    # 
    # \return \a self
    # 
    def assign(atom: Atom) -> Atom: pass

    ##
    # \brief 
    # \return 
    #
    def getAtoms() -> object: pass

    ##
    # \brief 
    # \return 
    #
    def getBonds() -> object: pass

    ##
    # \brief Returns a reference to the connected atom at index <em>idx</em>.
    # 
    # \param idx The index of the atom to return.
    # 
    # \return A reference to the connected atom at index <em>idx</em>. 
    # 
    # \throw Base.IndexError if <em>idx</em> is not in the range [0, getNumAtoms() - 1].
    # 
    # \note The order of connected atoms corresponds to the order of connecting bonds.
    # 
    def getAtom(idx: int) -> Atom: pass

    ##
    # \brief Tells whether this atom and the argument atom are connected by a bond.
    # 
    # \param atom The atom to test for adjacency.
    # 
    # \return <tt>True</tt> if the argument atom is connected to this atom, and <tt>False</tt> otherwise.
    # 
    def containsAtom(atom: Atom) -> bool: pass

    ##
    # \brief Returns the index of the specified connected atom.
    # 
    # \param atom The connected atom for which to return the index.
    # 
    # \return The zero-based index of the specified atom. 
    # 
    # \throw Base.ItemNotFound if the specified atom is not connected to this atom.
    # 
    def getAtomIndex(atom: Atom) -> int: pass

    ##
    # \brief Returns the number of connected atoms.
    # 
    # Equivalent to getNumBonds().
    # 
    # \return The number of connected atoms.
    # 
    def getNumAtoms() -> int: pass

    ##
    # \brief 
    # \param func 
    #
    def orderAtoms(func: ForceField.InteractionFilterFunction2) -> None: pass

    ##
    # \brief 
    # \param idx 
    # \return 
    #
    def getEntity(idx: int) -> Entity3D: pass

    ##
    # \brief 
    # \return 
    #
    def getNumEntities() -> int: pass

    ##
    # \brief Returns a reference to the incident bond at index <em>idx</em>.
    # 
    # \param idx The index of the incident bond to return.
    # 
    # \return A reference to the incident bond at index <em>idx</em>. 
    # 
    # \throw Base.IndexError if <em>idx</em> is not in the range [0, getNumBonds() - 1].
    # 
    # \note The order of incident bonds corresponds to the order of connected atoms.
    # 
    def getBond(idx: int) -> Bond: pass

    ##
    # \brief Tells whether the specified bond is incident to this atom.
    # 
    # \param bond The bond to test for incidence.
    # 
    # \return <tt>True</tt> if the argument bond is incident to this atom, and <tt>False</tt> otherwise.
    # 
    def containsBond(bond: Bond) -> bool: pass

    ##
    # \brief 
    # \param func 
    #
    def orderBonds(func: BoolBond2Functor) -> None: pass

    ##
    # \brief Returns the index of the specified incident bond.
    # 
    # \param bond The bond for which to return the index.
    # 
    # \return The zero-based index of the specified bond. 
    # 
    # \throw Base.ItemNotFound if the specified bond is not incident to this atom.
    # 
    def getBondIndex(bond: Bond) -> int: pass

    ##
    # \brief Returns the number of incident bonds.
    # 
    # Equivalent to getNumAtoms().
    # 
    # \return The number of incident bonds.
    # 
    def getNumBonds() -> int: pass

    ##
    # \brief Returns the result of the membership test operation <tt>atom in self</tt>.
    # \param atom The value to test for membership.
    # \return The result of the membership test operation.
    # 
    def __contains__(atom: Atom) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>bond in self</tt>.
    # \param bond The value to test for membership.
    # \return The result of the membership test operation.
    # 
    def __contains__(bond: Bond) -> bool: pass

    ##
    # \brief Returns the result of the membership test operation <tt>key in self</tt>.
    # \param key The value to test for membership.
    # \return The result of the membership test operation.
    # 
    def __contains__(key: Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __getitem__(key: Base.LookupKey) -> Base.Any: pass

    ##
    # \brief 
    # \param key 
    # \param value 
    #
    def __setitem__(key: Base.LookupKey, value: Base.Any) -> None: pass

    ##
    # \brief 
    # \param key 
    # \return 
    #
    def __delitem__(key: Base.LookupKey) -> bool: pass

    ##
    # \brief 
    # \return 
    #
    def __len__() -> int: pass

    molecule = property(getMolecule)

    index = property(getIndex)

    atoms = property(getAtoms)

    bonds = property(getBonds)
