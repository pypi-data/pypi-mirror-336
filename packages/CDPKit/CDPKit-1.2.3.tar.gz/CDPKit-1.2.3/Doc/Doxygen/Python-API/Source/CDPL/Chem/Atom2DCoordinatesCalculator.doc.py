#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Atom2DCoordinatesCalculator.
# 
class Atom2DCoordinatesCalculator(Boost.Python.instance):

    ##
    # \brief Constructs the <tt>Atom2DCoordinatesCalculator</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Constructs the <tt>Atom2DCoordinatesCalculator</tt> instance and calculates 2D-coordinates for the atoms of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to calculate 2D-coordinates.
    # \param coords An array containing the calculated 2D-coordinates. The coordinates are stored in the same order as the atoms appear in the atom list of the molecular graph (i.e. the coordinates of an atom are accessible via its index).
    # 
    def __init__(molgraph: MolecularGraph, coords: Math.Vector2DArray) -> None: pass

    ##
    # \brief Returns the numeric identifier (ID) of the wrapped C++ class instance.
    # 
    # Different Python \c %Atom2DCoordinatesCalculator instances may reference the same underlying C++ class instance. The commonly used Python expression
    # <tt>a is not b</tt> thus cannot tell reliably whether the two \c %Atom2DCoordinatesCalculator instances \e a and \e b reference different C++ objects. 
    # The numeric identifier returned by this method allows to correctly implement such an identity test via the simple expression
    # <tt>a.getObjectID() != b.getObjectID()</tt>.
    # 
    # \return The numeric ID of the internally referenced C++ class instance.
    # 
    def getObjectID() -> int: pass

    ##
    # \brief Calculates 2D-coordinates for the atoms of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to calculate 2D-coordinates.
    # \param coords An array containing the calculated 2D-coordinates. The coordinates are stored in the same order as the atoms appear in the atom list of the molecular graph (i.e. the coordinates of an atom are accessible via its index).
    # 
    def calculate(molgraph: MolecularGraph, coords: Math.Vector2DArray) -> None: pass

    objectID = property(getObjectID)
