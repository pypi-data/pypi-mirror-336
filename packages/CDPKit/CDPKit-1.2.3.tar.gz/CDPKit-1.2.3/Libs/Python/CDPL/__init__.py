##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
