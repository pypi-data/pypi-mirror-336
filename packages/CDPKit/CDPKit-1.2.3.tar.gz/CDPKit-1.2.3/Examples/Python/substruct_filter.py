#!/bin/env python

##
# substruct_filter.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import argparse

import CDPL.Chem as Chem


def filterMolecules() -> None:
    args = parseArgs()
    
    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(args.in_file) 

    # create writer for the output of matching molecules (format specified by file extension)
    writer = Chem.MolecularGraphWriter(args.out_file) 

    # parse the substructure SMARTS pattern
    try:
        sub_srch_ptn = Chem.parseSMARTS(args.smarts_ptn)
    except Exception as e:
        sys.exit('Error: parsing of SMARTS pattern failed: %s' % str(e))

    # create and initialize an instance of the class Chem.SubstructureSearch that
    # implements the substructure searching algorithm
    substr_srch = Chem.SubstructureSearch(sub_srch_ptn)

    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    i = 1
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol):
            # compose a simple molecule identifier
            mol_id = Chem.getName(mol).strip() 

            if mol_id == '':
                mol_id = '#' + str(i) # fallback if name is empty
            else:
                mol_id = '\'%s\' (#%s)' % (mol_id, str(i))

            if not args.quiet:
                print('- Searching for a matching substructure in molecule %s...' % mol_id)

            try:
                Chem.initSubstructureSearchTarget(mol, False)

                if substr_srch.mappingExists(mol):
                    if not args.quiet:
                        print(' -> substructure found, forwarding molecule to output file')

                    # output the matched molecule                 
                    if not writer.write(mol):   
                        sys.exit('Error: output of molecule failed')

                elif not args.quiet: 
                    print(' -> substructure not found')
                        
            except Exception as e:
                sys.exit('Error: substructure search or output of molecule %s failed: %s' % (mol_id, str(e)))

            i += 1
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed: ' + str(e))

    writer.close()
    sys.exit(0)
        
def parseArgs() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description='Writes input molecules that match the specified SMARTS substructure pattern to an output file.')

    parser.add_argument('-i',
                        dest='in_file',
                        required=True,
                        metavar='<file>',
                        help='Molecule input file')
    parser.add_argument('-o',
                        dest='out_file',
                        required=True,
                        metavar='<file>',
                        help='Molecule output file')
    parser.add_argument('-p',
                        dest='smarts_ptn',
                        required=True,
                        metavar='<SMARTS>',
                        help='SMARTS pattern describing the substructure to search for')
    parser.add_argument('-q',
                        dest='quiet',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Disable progress output (default: false)')
    
    return parser.parse_args()

if __name__ == '__main__':
    filterMolecules()
