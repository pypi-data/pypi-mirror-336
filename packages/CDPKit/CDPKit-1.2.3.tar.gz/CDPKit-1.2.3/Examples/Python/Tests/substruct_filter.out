Nc1cnc(cn1)c1ccc(C2CCC2)c(Oc2cc(N)nc(Cc3ccccc3)n2)c1F
N1C(\C(C(c2ccc(cc12)C)=O)=N/Nc2c(cccc2)Cl)=O
C([C@H]1[C@@](C1)(C(N(CC)CC)=O)c1ccccc1)N
C([C@H]1[C@@](C1)(C(N(CC)CC)=O)c1ccccc1)N
C(CN([C@@H](c1ccccc1)C)C)=C=CC
C(CN([C@@H](c1ccccc1)C)C)=C=CC
[N](c1ccc(NC(=O)CC[C@@H](C(=O)O)N)cc1)(=O)=O
c12ccc(cc2nc([nH]1)[S@@](Cc1ncc(c(c1C)OC)C)=O)OC
c1(cccc(c1Nc1ccccc1CC(=O)OCCNC(=O)[C@H]1O[C@H]([C@@H]([C@H]([C@@H]1O[C@@H]1O[C@@H]([C@H]([C@@H]([C@H]1NC(=O)C)O)O)CO)O)O)O[C@H]1[C@@H]([C@H](O[C@H]([C@@H]1NC(=O)C)O[C@@H]1[C@H](O[C@H]([C@@H]([C@H]1O)O)O)C([O-])=O)CO)O)Cl)Cl.[Na+]
CC(C)N=C1C=C2N(c3ccc(Cl)cc3)c3ccccc3N=C2C=C1Nc1ccc(Cl)cc1
O1[C@@H]([C@H]([C@@H](O)C)C)[C@@H]1C[C@@H]1[C@H]([C@@H](O)[C@@H](OC1)CC(CC(c1ccc(cc1)N(C)C)=O)=O)O
c1(Cl)[nH]c2nc1c1c(NC(CC\C=C\C[C@@H]2NC(c2c(cc(cc2F)C)F)=O)=O)cc(cc1)NC(OC)=O  
Cc1cc(Nc2c3cc(NC4=NC(C)(C)CO4)ccc3ncn2)ccc1Oc1cc2ncnn2cc1
O(c1c(cc([N+]([O-])=O)cc1)C#N)C(=O)c1ccc(NC)cc1
c1cc(C(N2[C@H]3CC[C@@H]2C2CNCC23)=O)nc3ccc(cc13)Oc3ccc(cn3)C(F)(F)F BDBM250863
O=S(=O)(NC([C@]1(C[C@H]1[C@H]1CC1)NC([C@H]1N(C([C@@H](NC(=O)O[C@@H]2C[C@H]3[C@@H](C2)C3)[C@](C)(C)C)=O)C[C@@H](C1)Oc1ncc(c3ccc(cc13)Cl)OC)=O)=O)C3CC3  
C1C[C@@H]2C[C@@H](c3cc(O)ccc3)C[C@@H]1N2CCCOc2ccccc2  
CC[O+]1[CH-]C=Cc2c(NCc3ccccc3)nc(Cc3ccccc3)nc21 BDBM163075
N1(CCN(CC1)c1ccc(cc1)OC[C@@H]1O[C@](OC1)(c1c(Cl)cc(cc1)Cl)Cn1cncn1)c1ccc(cc1)N1C=NN(C1=O)C(C)CC
Cc1ccc(NC(=O)c2ccnc(c2)N2CCCC2)cc1c1ccc(cc1)C(=O)NCC1CC1
c1c(cccc1)C(=O)O[Na]
O[C@H](C(O)=O)[C@H](O)C(=O)O.Oc1c(O)ccc(c1)[C@@H](O)CNC
O=C1N(NC(N)=C1C(Cc1c(cc([N+](=O)[O-])cc1)C)(O)C)c1ccccc1
c12ccc(cc2nc([nH]1)S(Cc1ncc(c(c1C)OC)C)=O)OC  
C1(c2ccccc2)(OCC[N+](C)(C[18F])C1)O.c1ccc(cc1O)c1cc(cc(c1)O)O
O1[C@@H]([C@H]([C@@H](O)C)C)[C@@H]1C[C@@H]1[C@H]([C@@H](O)[C@@H](OC1)CC(CC(c1ccc(cc1)N(C)C)=O)=O)O
c1c[n+]2c3c(c1NCCCCCNc1cc[n+](c4c1cccc4)CC=CC2)cccc3
c1(C(C=Cc2ccccc2)=O)c(N)cccc1
O=C1N(NC(N)=C1C(Cc1c(cc([N](=O)=O)cc1)C)(O)C)c1ccccc1
c12c(Nc3c(cc(cc3)NS(=O)(C)=O)OC)c3c(nc1cccc2)cc(cc3)N=[N]=N
C1C=CN2C(C=1)=C(C(N(C2=O)CCCCN2CCC(CC2)c2c[nH]c1c2[F]=C(C=C1)F)=O)c1ccc(cc1)F
BrC(C(=O)\N=[N]1=C2C(=Nc3c\1cccc3)c3c1c2cccc1ccc3)C
C1([N+](=C(N(C1(C)C)O)c1ccc(cc1)OCC(=O)N[C@H](C(O)=O)CCCNC(=N)N)[O-])(C)C
Cn1cc(C=C2C(=O)Nc3cccnc23)c3ccccc13
c12ccc(cc1[C@]13[C@@H]([C@@H](N(CC1)C)C2)CCCC3)OC
[123I]c1ccc(cc1)CCCCCCCCCCCCCCC(O)=O
c1(c(c(n(c1C(C)C)CC[C@H](C[C@H](CC(=O)[O-])O)O)c1ccc(F)cc1)c1ccccc1)C(=O)Nc1ccccc1.[Ca+2].c1(c(c(n(c1C(C)C)CC[C@H](C[C@H](CC(=O)[O-])O)O)c1ccc(cc1)F)c1ccccc1)C(=O)Nc1ccccc1
CC(Nc1ncc2ccc(cc2n1)C(=N[C@@H](c1ccc(c(c1)F)Cl)c1cnn(C)c1)O)[H]CO BDBM192334
C(c1ccccc1)C([NH3+])C.[Cl-]
C(c1ccccc1)C([NH3+])C
N(C(=[NH2+])N)CCCC(C(OC(C)C(OS(=O)(=O)[O-])c1cc(c(cc1)OS(=O)([O-])=O)COS(=O)(=O)[O-])=O)[N+](C)(C)C
CCCc1c2c(C(NC(c3cc(ccc3OCC)S(N3CCN(C)CC3)(=O)=O)=N2)=O)n(C)n1.C(C(=O)O)C(CC(=O)O)(C(=O)O)O
C1[C@H]2N3[C@H](O[C@@H]1CC2)CN2C(C3=O)=C(C(C(=C2)C(NCc2c(cc(cc2F)F)F)=O)=O)[O-].[Na+]
c1(c(oc(n1)CCC([O-])=O)c1ccccc1)c1ccccc1.[K+]
c12ccc(cc2nc([n-]1)[S+](Cc1ncc(c(c1C)OC)C)[O-])OC.c12ccc(cc2nc([n-]1)[S+](Cc1ncc(c(c1C)OC)C)[O-])OC.[Mg+2]
c12ccc(cc2nc([n-]1)[S+](Cc1ncc(c(c1C)OC)C)[O-])OC.c12ccc(cc2nc([n-]1)[S+](Cc1ncc(c(c1C)OC)C)[O-])OC.[Sr+2].O.O.O.O
CC(C)(c1c(c(c2cc3ccc(NS(C)(=O)=O)cc3cc2)cc(c1)N1C([N-]C(=O)C=C1)=O)OC)C.[Na+].O
C1(=C([C@@H]([C@H]2OC(c3ccccc3)([2H])OC2)OC1=O)O)O
C1(=C([C@@H]([C@H]2OC(c3ccccc3)([2H])OC2)OC1=O)O)O
COC1CN(CCCOc2ccc(F)cc2)CCC1NC(=O)c1cc(Cl)c(N)cc1OC.O
Cc1cc(no1)NS(=O)(=O)c1ccc(cc1)N.COc1cc(cc(c1OC)OC)Cc1cnc(nc1N)N
N1(N(C(=O)C(C1=O)CCCC)C)C.C(=O)(NCCN(CC)CC)COc1ccc(cc1)Cl.O.O
Cc1ccc(NC(=O)c2ccnc(c2)N2CCCC2)cc1c1ccc(cc1)C(=O)NCC1CC1
C(C(N)c1c(cc(cc1Cl)O)Cl)(N)c1c(Cl)cc(cc1Cl)O.S(=O)(=O)(O)O.[Pt].O